cmd/bedFormat/bedFormat.go

```diff
@@ -99,7 +133,64 @@ func bedFormat(s Settings) {
 		if s.ScaleNameFloat != 1 {
 			v.Name = fmt.Sprintf("%.8g", s.ScaleNameFloat*parse.StringToFloat64(v.Name))
 		}
-		bed.WriteBed(out, v)
+
+		if s.FdrAnnotation {
+			totalWindows++
+			if s.RawPValueAnnotationField >= len(v.Annotation) {
+				log.Fatalf("Error: rawPValueAnnotationField, %v, exceeds the length of the annotation slice in bed entry: %v.\n", s.RawPValueAnnotationField, len(v.Annotation))
+			}
+			if mapEntry, inMap = fdrMap[parse.StringToFloat64(v.Annotation[s.RawPValueAnnotationField])]; inMap {
+				mapEntry.Count++
+				mapEntry.RawPValue = parse.StringToFloat64(v.Annotation[s.RawPValueAnnotationField])
+				fdrMap[parse.StringToFloat64(v.Annotation[s.RawPValueAnnotationField])] = mapEntry
+			} else {
+				fdrMap[parse.StringToFloat64(v.Annotation[s.RawPValueAnnotationField])] = FdrConverter{Count: 1, Rank: 0, RawPValue: parse.StringToFloat64(v.Annotation[s.RawPValueAnnotationField]), AdjPValue: 0}
+			}
+
+			bed.WriteBed(tmp, v)
+		} else {
+			bed.WriteBed(out, v)
+		}
+	}
+
+	if s.FdrAnnotation {
+		err = tmp.Close()
+		exception.PanicOnErr(err)
+
+		var currRank int = 0
+		var fdrSlice []FdrConverter = make([]FdrConverter, 0)
+		for i := range fdrMap {
+			fdrSlice = append(fdrSlice, fdrMap[i])
+		}
+
+		sort.Slice(fdrSlice, func(i, j int) bool { return compareFdrConverter(fdrSlice[i], fdrSlice[j]) == -1 })
+
+		for i := len(fdrSlice) - 1; i >= 0; i-- {
+			currRank += fdrSlice[i].Count
+			fdrSlice[i].Rank = currRank
+
+			// note here I move from -log10 space to log10 space for readability
+			fdrSlice[i].RawPValue = fdrSlice[i].RawPValue * -1                                                          //from -log10p to log10p
+			fdrSlice[i].AdjPValue = fdrSlice[i].RawPValue + math.Log10(float64(totalWindows)/float64(fdrSlice[i].Rank)) //note that addition is logspace multiply
```

Comment:
Can I ask why adjPValue is calculated with rawPValue*(totalWindows)/(rank)? I think this means that the more total windows there are, the bigger adjPValue becomes, and the less likely it will be significant. I thought it would be the other way around?